torch/_inductor/runtime/coordinate_descent_tuner.py

```diff
@@ -54,29 +54,20 @@ def __init__(
         self.size_hints = size_hints
         self.inductor_meta = inductor_meta or {}
 
-    def get_xmax(self):
-        xmax = TRITON_MAX_BLOCK["X"]
-        if self.size_hints and len(self.size_hints) > 0:
-            xmax = min(xmax, self.size_hints[0])
-        return xmax
-
-    def get_ymax(self):
-        ymax = TRITON_MAX_BLOCK["Y"]
-        if self.size_hints and len(self.size_hints) > 1:
-            ymax = min(ymax, self.size_hints[1])
-        return ymax
-
-    def get_zmax(self):
-        zmax = TRITON_MAX_BLOCK["Z"]
-        if self.size_hints and len(self.size_hints) > 2:
-            zmax = min(zmax, self.size_hints[2])
-        return zmax
-
-    def get_rmax(self):
-        rmax = TRITON_MAX_BLOCK["R"]
-        if self.size_hints and len(self.size_hints) > 0:
-            rmax = min(rmax, self.size_hints[-1])  # the last one is for reduction
-        return rmax
+    def prefix_to_size_hint(self, prefix: str) -> Optional[int]:
+        size_hint_idx = {"X": 0, "Y": 1, "Z": 2, "R": -1}[prefix]
+
+        have_size_hint = (
+            self.size_hints is not None
+            and len(self.size_hints) > 0
+            and len(self.size_hints) > size_hint_idx
```

Comment:
Looks good overall, but size_hint_idx value '-1' need to be specially handled (i.e. wrap around).

Reply:
That one seems tricky. My thought here was, if `size_hint_idx == -1`, then we always have `len(self.size_hints) > size_hint_idx`. So in that case, this is just checking that we have any size hints at all.

Reply:
yea. Checking `len(self.size_hints) > 0` rather than `len(self.size_hints) > -1` looks more correct.